### PR TITLE
Client: AJAX spider enhancement

### PR DESCRIPTION
**Pull request by psiinon**
## Overview

* Add support for enhancing AJAX spider results
* Add more help pages
* Tweak cookie event handling

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by psiinon** [comment]
Still need to look into unit tests...

**Comment by psiinon** [comment]
Tests added - ready for review.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

File: addOns/client/src/main/javahelp/org/zaproxy/addon/client/resources/help/contents/client.html [hunk 1]
```diff
@@ -10,6 +10,8 @@ <H1>Client Side Integration</H1>
 
 This add-on adds a ZAP browser extension to both Firefox and Chrome when they are launched from ZAP.
 <p>
+The extensions steam significant browser based events back to ZAP, giving ZAP an insight into what is going on in the browser.
```
Comment (ricekot) [inline addOns/client/src/main/javahelp/org/zaproxy/addon/client/resources/help/contents/client.html:13]: steam -> stream

Comment (kingthorin) [reply addOns/client/src/main/javahelp/org/zaproxy/addon/client/resources/help/contents/client.html:13]: Doh, good catch

**Pull request merged**